packages/mongodb-chatbot-server/src/routes/responses/createResponse.test.ts

```diff
@@ -491,4 +586,68 @@ describe("POST /responses", () => {
       );
     });
   });
+
+  it("Should return 400 if user id has changed since the conversation was created", async () => {
+    const userId1 = "user1";
+    const userId2 = "user2";
+    const conversation =
+      await appConfig.conversationsRouterConfig.conversations.create({
+        initialMessages: [
+          {
+            role: "user",
+            content: "What is MongoDB?",
+            customData: { userId: userId1 },
+          },
+        ],
+      });
+
+    const previousResponseId = conversation.messages[0].id.toString();
+    const response = await makeCreateResponseRequest({
+      previous_response_id: previousResponseId,
+      user: userId2,
+    });
+
+    expect(response.statusCode).toBe(400);
+    expect(response.body.error).toEqual(
+      badRequestError(ERR_MSG.CONVERSATION_USER_ID_CHANGED)
+    );
+  });
+
+  it("Should return 400 if previous_response_id is provided but store is false", async () => {
+    const response = await makeCreateResponseRequest({
+      previous_response_id: "123456789012123456789012",
+      store: false,
+    });
+
+    expect(response.statusCode).toBe(400);
+    expect(response.body.error).toEqual(
+      badRequestError(ERR_MSG.STORE_NOT_SUPPORTED)
+    );
+  });
+
+  it("Should return 400 if conversation store flag is false but store is true", async () => {
```

Comment:
```suggestion
  it("Should return 400 if `Conversation.stor: false` but request `store: true`", async () => {
```

nit for clarity

Reply:
That's actually already handled by this case `"Should return 400 if previous_response_id is provided but store is false"`. It won't even attempt to look up the conversation under those conditions. You can't get a convo without passing a `previous_response_id`, and you can't pass a `previous_response_id` and `store:false`